### PR TITLE
refactor(ui): add session guard to ui page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add session guard to ui page

- add session check guard to UI page request only
- redirect to the unknonwn-session page if no session is found
image

## Testing

test locally 
ut case added

@logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/middleware/koa-client-session-guard.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  it('should redirect if seesion not found', async () => {
+    const provider = new Provider('');
+
+    (provider.interactionDetails as jest.Mock).mockRejectedValue(new Error('session not found'));
+    const ctx = createContextWithRouteParameters({
+      url: '/sign-in',
+    });
+    await koaClientSessionGuard(provider)(ctx, next);
+    expect(ctx.redirect).not.toBeCalled();
```
Comment (demonzoo) [inline packages/core/src/middleware/koa-client-session-guard.test.ts:75]: should call

**Review by demonzoo — CHANGES_REQUESTED** [review]
Please refine the last test case. It says `should redirect`, but in the end it expect redirect `not to be called`. I'm confused.

File: packages/core/src/middleware/koa-client-session-guard.test.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-client-session-guard.test.ts]: updated, caused by wrong mock dist file name

Comment (simeng-li) [reply packages/core/src/middleware/koa-client-session-guard.test.ts]: @demonzoo please check

File: packages/core/src/middleware/koa-client-session-guard.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    (provider.interactionDetails as jest.Mock).mockRejectedValue(new Error('session not found'));
+    const ctx = createContextWithRouteParameters({
+      url: `${sessionNotFoundPath}`,
+    });
+    await koaClientSessionGuard(provider)(ctx, next);
+    expect(ctx.redirect).not.toBeCalled();
+  });
+
+  it('should redirect if seesion not found', async () => {
```
Comment (demonzoo) [inline packages/core/src/middleware/koa-client-session-guard.test.ts:67]: typo: session

**Review by demonzoo — APPROVED** [review]

File: packages/core/src/middleware/koa-client-session-guard.test.ts (hunk 2 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-client-session-guard.test.ts]: updated

**Pull request merged**